agent/time.go

```diff
@@ -0,0 +1,13 @@
+package agent
+
+import (
+	"math/rand"
+	"time"
+)
+
+// fuzzInterval returns a duration that is gs
+func fuzzInterval(interval time.Duration) time.Duration {
+	val := (rand.Float64() * float64(interval)) + interval
```

Comment:
write a test (e.g., a hundred iterations of `fuzzInterval(15 * time.Second)` always returns a value 15 seconds < value < 30 seconds)